src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -388,19 +415,32 @@ public Response corsPreflightQuery(@HeaderParam("Access-Control-Request-Headers"
 	@GET
 	@Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
 	@Path(QUERY_URL)
-	public Response query(@QueryParam("query") String json) throws Exception
+	public Response query(@Context HttpHeaders httpHeaders, @QueryParam("query") String json) throws Exception
 	{
-		return get(json);
+		return get(httpHeaders, json);
 	}
 
 	@POST
 	@Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
 	@Path(QUERY_URL)
-	public Response get(String json) throws Exception
+	public Response get(@Context HttpHeaders httpHeaders, String json) throws Exception
 	{
 		checkNotNull(json);
 		logger.debug(json);
 
+		ScopeManager scopeManager = tracer.scopeManager();
+		SpanContext spanContext = tracer.extract(Format.Builtin.HTTP_HEADERS, new HttpHeadersCarrier(httpHeaders.getRequestHeaders()));
+		Tracer.SpanBuilder spanBuild = tracer.buildSpan(QUERY_URL).withTag("Name",QUERY_URL).withTag(Tags.SPAN_KIND.getKey(),Tags.SPAN_KIND_SERVER);
```

Comment:
Same comment for ``"Name"``

Reply:
I think we are missing some important OT tags here. I want to be able to search by ``metricname`` and may be KairosDB tags (e.g ``stack_name``, ``application_id``, ``key`` etc...) This info I believe exists in the json payload so may be extracted in the underlying layers, but needed nevertheless.

Reply:
Some explanation around why these tags are required would be helpful in understanding this requirement. AFAIK IMHO we need to know the checkid's and associated dashboard from where the queries are submitted. Nevertheless these can be included just means more work -> more time for the PR.

Reply:
Not having the ``metricname`` or ``check_id`` makes all this instrumentation useless, since it can help us in the search and in diagnosis. Regarding the associated ``dashboard``, I don't think KairosDB should be concerned with what a Dashboard is, it should provide all necessary info that it already knows, concerned with. KairosDB should only assume API calls from some component, regardless of the nature of the calling component (e.g. ZMON worker can call KairosDB and it is not a ``dashboard``).

Regarding the more time for the PR, not sure if that is a problem since the PR is already in progress for more than 3 months.